tensorflow/core/kernels/tensor_forest/resources.cc

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/kernels/tensor_forest/resources.h"
+#include "tensorflow/core/kernels/boosted_trees/boosted_trees.pb.h"
+#include "tensorflow/core/platform/protobuf.h"
+
+namespace tensorflow {
+
+const float TensorForestTreeResource::get_prediction(
+    const int32 id, const int32 dimension) const {
+  return decision_tree_->nodes(id).leaf().vector().value(dimension);
+};
+
+const int32 TensorForestTreeResource::TraverseTree(
+    const TTypes<float>::ConstMatrix* dense_data,
+    const int32 example_id) const {
```

Comment:
lets put all the const first, and then the arguments that can change (example_id, dense_data). That's a good convention to have